### PR TITLE
Add support for Ruby 3.0 & 3.1

### PR DESCRIPTION
**Pull request by HeyNonster**
Add support for Ruby 3.0 & 3.1.

The `sorted_set` gem was removed from the standard library so we need to add it to the gemfiles explicitly.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,14 +14,26 @@ jobs:
...
           - rails4.2
           - rails5.1
           - rails5.2
           - rails6.0
           - rails6.1
+          - rails7.0
         exclude:
           - {ruby: '2.7', gemfile: rails4.2}
+          - {ruby: '2.6', gemfile: rails7.0}
+          - {ruby: '3.0', gemfile: rails4.2}
```
Comment (bquorning) [inline .github/workflows/ci.yml:29]: Something is odd with the order of this list.

File: gemfiles/rails6.1.gemfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
 eval_gemfile 'common.rb'
 
 gem 'rails', '~> 6.1.0'
-gem 'genspec', github: 'bquorning/genspec', branch: 'rails-6'
+gem 'genspec', github: 'zendesk/genspec', branch: 'rails-7'
+gem 'sorted_set'
```
Comment (bquorning) [inline gemfiles/rails6.1.gemfile:5]: I think we need to add `sorted_set` as a runtime dependency, and also `require "sorted_set"` lib/curly/presenter.rb

File: .github/workflows/ci.yml (hunk 1 above)
Comment (HeyNonster) [reply .github/workflows/ci.yml:29]: Fixed!

File: curly-templates.gemspec [hunk 3]
```diff
@@ -22,6 +22,8 @@ Gem::Specification.new do |s|
 
   s.add_dependency("actionpack", [">= 4.2", "< 7.1"])
 
+  s.add_runtime_dependency("sorted_set")
```
Comment (bquorning) [inline curly-templates.gemspec:25]:
```suggestion
  s.add_dependency("actionpack", [">= 4.2", "< 7.1"])
  s.add_dependency("sorted_set")
```

File: gemfiles/rails6.1.gemfile.lock [hunk 4]
```diff
@@ -101,13 +102,9 @@ GEM
...
     minitest (5.14.3)
     nio4r (2.5.4)
-    nokogiri (1.11.1)
+    nokogiri (1.11.7)
       mini_portile2 (~> 2.5.0)
       racc (~> 1.4)
-    nokogiri (1.11.1-x86_64-darwin)
-      racc (~> 1.4)
-    nokogiri (1.11.1-x86_64-linux)
-      racc (~> 1.4)
```
[the remark this answers is not in the document]
Comment (HeyNonster) [reply gemfiles/rails6.1.gemfile.lock:110]: I believe that it's related to this: https://github.com/sparklemotion/nokogiri/issues/2397

Some versions added support for `Ruby 3.1` but the precompiled native gems did not have the same compatibility.

**Review by bquorning — APPROVED** [review]

**Pull request merged**